springfox-schema/src/main/java/springfox/documentation/schema/property/OptimizedModelPropertiesProvider.java

```diff
@@ -218,6 +220,58 @@ public boolean apply(Annotation input) {
 
   }
 
+  private boolean shouldBeIncludeDueToJsonView(BeanPropertyDefinition jacksonProperty,
+                                               ModelContext givenContext) {
+
+    Class classToRestrictOn = null;
+    JsonView jsonView = givenContext.getJsonView();
+    if (jsonView != null) {
+      classToRestrictOn = jsonView.value()[0];
+    }
+    List<Class> expectedClasses = getAllClasses(classToRestrictOn, new ArrayList<Class>());
+
+    if (expectedClasses.isEmpty()) {
+      return true;
+    } else {
+      Class[] views = jacksonProperty.findViews();
+      if (views != null) {
+        List<Class<?>> fieldViews = Arrays.asList(jacksonProperty.findViews());
+        for (Class eachView : fieldViews) {
+          if (expectedClasses.contains(eachView)) {
+            return true;
+          }
+        }
+      }
+    }
+
+    return false;
+  }
+
+  private static List<Class> getAllClasses(Class clazz, List<Class> collectedClasses) {
+
+    // Already recurse on it.
+    if (clazz == null || collectedClasses.contains(clazz)) {
+      return collectedClasses;
+    }
+
+    collectedClasses.add(clazz);
+
+    Class[] nextClasses;
+
+    // If the class is an interface, we get all extended interfaces.
+    if (clazz.isInterface()) {
```

Comment:
This now support is the given class to the json view is an interface (better for multiple extends) or an class